gds_write/def2stream.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+
+import klayout.db as db
+import re
+import json
+import copy
+import sys
+import os
+import argparse
+
+parser = argparse.ArgumentParser(description="GDS writer script")
+parser.add_argument("-d", "--design-name",    required=True,  help="Name of the top module in the design")
+parser.add_argument("-i", "--input-def",      required=True,  help="Path to def file of the implemented design after detailed routing step")
+parser.add_argument("-t", "--tech-file",      required=True,  help="Path to KLayout technology file, a mapping from LEF/DEF to GDS.")
+parser.add_argument("-o", "--out",            required=True,  help="Path to output GDS file")
+parser.add_argument("-g", "--additional-gds", required=False, help="Paths to additional input GDS files", default="", nargs="*")
+parser.add_argument("-f", "--fill-config",    required=False, help="Path to JSON rule file for metal fill during chip finishing", default="")
```

Comment:
curious why we don't use openroad for that?
https://openroad.readthedocs.io/en/latest/main/src/fin/README.html

Reply:
We can do that. I added `density_fill` stage right after `detailed_routing`